tests/test_end_to_end.py

```diff
@@ -53,49 +53,95 @@ def nbstripout_exe():
 
 
 @pytest.mark.parametrize("input_file, expected_file, args", TEST_CASES)
-def test_end_to_end_stdin(input_file: str, expected_file: str, args: List[str]):
+@pytest.mark.parametrize("verify", (True, False))
+def test_end_to_end_stdin(input_file: str, expected_file: str, args: List[str], verify: bool):
     with open(NOTEBOOKS_FOLDER / expected_file, mode="r") as f:
         expected = f.read()
 
     with open(NOTEBOOKS_FOLDER / input_file, mode="r") as f:
-        pc = run([nbstripout_exe()] + args, stdin=f, stdout=PIPE, universal_newlines=True)
+        input_ = f.read()
+
+    with open(NOTEBOOKS_FOLDER / input_file, mode="r") as f:
+        args = [nbstripout_exe()] + args
+        if verify:
+            args.append("--verify")
+        pc = run(args, stdin=f, stdout=PIPE, universal_newlines=True)
         output = pc.stdout
 
-    assert output == expected
+    if verify:
+        # When using stin, the dry run flag is disregarded.
+        if input_ != expected:
+            assert pc.returncode == 1
+        else:
+            assert pc.returncode == 0
```

Comment:
```suggestion
        assert pc.returncode == 1 if input_ != expected else 0
```

Reply:
Note: This the tests really hard to figure out and I partially reverted the change bc it is equivalent to ...

```
>>> (pc.returncode == 1) if input_ != expected else (0)
```

instead of

```
>>> pc.returncode == (1 if input_ != expected else 0)
```